conjure-go-client/httpclient/client_params.go

```diff
@@ -542,13 +542,39 @@ func WithErrorDecoder(errorDecoder ErrorDecoder) ClientParam {
 // WithBasicAuth sets the request's Authorization header to use HTTP Basic Authentication with the provided username and
 // password.
 func WithBasicAuth(user, password string) ClientOrHTTPClientParam {
-	return WithMiddleware(&basicAuthMiddleware{provider: func(ctx context.Context) (BasicAuth, error) {
-		return BasicAuth{User: user, Password: password}, nil
-	}})
+	return WithMiddleware(MiddlewareFunc(func(req *http.Request, next http.RoundTripper) (*http.Response, error) {
+		setBasicAuth(req.Header, user, password)
+		return next.RoundTrip(req)
+	}))
```

Comment:
```suggestion
	return WithBasicAuthProvider(BasicAuthProvider(func(context.Context) (BasicAuth, error) {
		return BasicAuth{User: user, Password: password}, nil
	}))
```

Reply:
nit, but I think that this suggested construction would technically reduce code reuse/ensure that there's only one code path that actually sets the basic auth